### DISCUSSION TITLE
No connection adapters were found(maybe https issue)

### DISCUSSION THREAD

**Discussion by seungtaek** [netbox-community/netbox#11057] [Q&A]
Hello guys, 
I got some error when I connect to nebox using python,
I tried to fix it but no luck
could you give me some advice?

> import requests
import json

API_TOKEN = "xxxxx"
HEADERS = {'Authorization': f'Token {API_TOKEN}',
           'Content-Type': 'application/json', 'Accept': 'application/json'}
NB_URL = "'https://netbox.xxxx.com"

device_name = "ARISTA-7050SX-64"

requests_url = f"{NB_URL}/api/dcim/interfaces/?device={device_name}"
interfaces_parameters = {
    "device" : 322,
    "name" : "test",
    "description" : "Ethernet0/51",
    "type" : "virtual",
    "enable" : True,
    "mode" : "access"
}

requests.post(requests_url, headers = HEADERS, json=interfaces_parameters)

Error messages 

requests.exceptions.InvalidSchema: No connection adapters were found for "'https://netbox.xxxx.com/api/dcim/interfaces/?device=ARISTA-7050SX-64"

I changed the token and site name little. I think because of https certificate.
when I am using pynetbox, I could connect it without issue. 
How do i have to change my code? I read a netbox doc official but didn't mention about certificate.

Thank in advance

**Comment by kkthxbye-code** [comment]
> NB_URL = "'https://netbox.xxxx.com/"

Remove the extra '

**Reply by seungtaek** [reply]
> > NB_URL = "'https://netbox.xxxx.com/"
> 
> Remove the extra '

Thanks man, but I got a real error now :)

requests.exceptions.SSLError: HTTPSConnectionPool(host='netbox.xxxx.com', port=443): Max retries exceeded with url: /api/dcim/interfaces/?device=ARISTA-7050SX-64 (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:997)')))

any advice for this? I think i have to no activate certificate verification..but I couldn't find a source in doc

**Reply by kkthxbye-code** [reply]
>requests.post(requests_url, headers = HEADERS, json=interfaces_parameters)

>requests.post(requests_url, headers = HEADERS, json=interfaces_parameters, verify=False)

**Reply by seungtaek** [reply]
> > requests.post(requests_url, headers = HEADERS, json=interfaces_parameters)
> 
> > requests.post(requests_url, headers = HEADERS, json=interfaces_parameters, verify=False)

Thanks man!!!
